litemapy/schematic.py

```diff
@@ -503,28 +541,186 @@ def __repr__(self):
 
     def __getitem__(self, key):
         return self.__properties[key]
-    
+
     def __len__(self):
         return len(self.__properties)
 
+
 class Entity:
 
-    def __init__(self):
-        pass #TODO
+    def __init__(self, str_or_nbt):
+
+        if isinstance(str_or_nbt, str):
+            self._data = List(Compound)([Compound({'id': String(str_or_nbt)})])
+        else:
+            self._data = str_or_nbt
+
+        keys = self._data.keys()
+        if 'id' not in keys:
+            self._data['id'] = String('minecraft:empty')
```

Comment:
Is there any scenario when a valid litematic would contain an entity with no ID ? How does the game behave in such a case ?
If it's not supposed to happen, then maybe raising an exception would be better, to avoid creating a litematic could not load.

Reply:
Same comment for the other keys

Reply:
I think this problem I the reason why I implemented decoding from nbt in a separate method and not in the constructor.

Reply:
That's exactly what I was asking myself. I thought about an exception but wasn't sure how to implement it correctly in Python so II thought I'd give a default value if somehow someone created an NBT tag themselves that was missing id etc. But yeah, not sure how Minecraft will react to "minecraft:empty". I will look into exceptions a bit more.

Reply:
> Same comment for the other keys

I think default values for the other keys make sense. That way, users can create an Entity with only an id.

Reply:
Yes you are right, 0,0,0 is a valid default value.

Reply:
I added an exception that is raised if 'id' is missing and let's the user know what's going wrong (can be used for other missing keys as well).